adaptive/runner.py

```diff
@@ -347,62 +367,30 @@ class BlockingRunner(BaseRunner):
 
     """
 
-    def __init__(
-        self,
+    if inspect.iscoroutinefunction(learner.function):
+        raise ValueError("Coroutine functions can only be used " "with 'AsyncRunner'.")
+    warnings.warn(
+        "adaptive.BlockingRunner is deprecated, and will be removed in a "
+        "future version of adaptive.\n"
+        "Replace all uses of adaptive.BlockingRunner(...) with "
+        "asyncio.run(adaptive.Runner(...))",
+        DeprecationWarning,
+    )
+
+    r = AsyncRunner(
         learner,
         goal,
-        *,
-        executor=None,
-        ntasks=None,
-        log=False,
-        shutdown_executor=False,
-        retries=0,
-        raise_if_retries_exceeded=True,
-    ):
-        if inspect.iscoroutinefunction(learner.function):
-            raise ValueError(
-                "Coroutine functions can only be used " "with 'AsyncRunner'."
-            )
-        super().__init__(
-            learner,
-            goal,
-            executor=executor,
-            ntasks=ntasks,
-            log=log,
-            shutdown_executor=shutdown_executor,
-            retries=retries,
-            raise_if_retries_exceeded=raise_if_retries_exceeded,
-        )
-        self._run()
-
-    def _submit(self, x):
-        return self.executor.submit(self.learner.function, x)
+        executor=executor,
+        ntasks=ntasks,
+        log=log,
+        shutdown_executor=shutdown_executor,
+        retries=retries,
+        raise_if_retries_exceeded=raise_if_retries_exceeded,
+    )
 
-    def _run(self):
-        first_completed = concurrent.FIRST_COMPLETED
+    asyncio.run(r)
```

Comment:
This is Python 3.7 only :(

Reply:
but `self.ioloop.run_until_complete(r)` should do the same.

Reply:
We could also bump the version requirement

Reply:
Seems good to me!

Reply:
```suggestion
    r.ioloop.run_until_complete(r.task)
```